config/software/crystal.rb

```diff
@@ -42,7 +42,7 @@
   command "mkdir .build", env: env
   command "echo #{Dir.pwd}", env: env
   command "cp #{Dir.pwd}/crystal-#{ohai['os']}-#{ohai['kernel']['machine']} .build/crystal", env: env
-  command "bin/crystal build src/compiler/crystal.cr --release -o #{output_bin} -D without_openssl -D without_zlib", env: env
+  command "bin/crystal build src/compiler/crystal.cr --release#{" --no-debug" if _32_bit? } -o #{output_bin} -D without_openssl -D without_zlib", env: env
```

Comment:
If we don't use `--no-debug`, we hit the 4GB memory limit for 32 bits systems - we should cross-compile, then.